docs/apparmor.md

```diff
@@ -0,0 +1,139 @@
+trust, a Mandatory Access Control system is useful for limiting just how much
+of your local filesystem an attacker can potentially grab if you are using
+wkhtmltopdf on untrusted content. AppArmor is installed and will work more or
+less out of the box on:
+
+* [Ubuntu](https://wiki.ubuntu.com/AppArmor) and derivatives
+* [SuSE/SLES](https://documentation.suse.com/de-de/sles/15-GA/html/SLES-all/part-apparmor.html) and derivatives
+* [Debian](https://wiki.debian.org/AppArmor/HowToUse) and derivatives
+
+RedHat systems do **not** use AppArmor and instead relies entirely on SELinux.
+
+### AppArmor for using wkhtmltopdf on untrusted files
+AppArmor provides a kernel-based method to limit what wkhtmltopdf is allowed
+to do, and what files it can access. While the `--disable-local-file-access`
+option will block access to the local filesystem, an attacker able to exploit
+a CVE in the prebuilt binaries may be able to bypass this setting; which is
+where AppArmor comes into play. AppArmor provides a secure backstop for
+preventing local file access outside of approved locations, as well as preventing
+it from executing commands should an attacker find a way to do that.
+
+#### Enable AppArmor if it isn't already
+To check if AppArmor is running on your Ubuntu, SuSE/SLES, or Debian system
+you can run the command:
+
+    systemctl status apparmor
+
+This should show it both *loaded* and *active*.
+
+If this command shows `Loaded: not-found`, then AppArmor isn't even installed.
+RedHat and Fedora systems do not use AppArmor at all, and use SELinux instead,
+so this procedure will not work at all with them.
+
+If this shows `Loaded: inactive`, you can enable AppArmor on boot through:
+
+    systemctl enable apparmor
+    systemctl start apparmor
+
+Running the status command again should show you both loaded and active.
+
+### Installing the wkhtmltopdf AppArmor profile
+The two step process for installing the profile is:
+
+1. Put the profile file into `/etc/apparmor.d`
+1. Reload the apparmor profiles through `systemctl reload apparmor`
+
+The profile config-file needs to be customized for your specific usage. The
+example below contains the base allows that wkhtmltopdf will need, plus two
+example rules for an example application. Please add rules for your specific
+use-case in the `# Working paths for your application:` section.
+
+File: /etc/apparmor.d/usr.local.bin.wkhtmltopdf
+```
+#include <tunables/global>
+
+/usr/local/bin/wkhtmltopdf {
+  # Allow base system access.
+  #include <abstractions/base>
+
+  # Allows:
+  #  - Basic network connectivity (TCP/UDP)
+  #  - Nameservice access (the whole resolver chain)
+  #  - Ability to interact with PAM
+  #  - Ability to interact with name-services like nis, ldap, winbind, kerberos
+  # Remove if you don't need network access.
+  #include <abstractions/nameservice>
+
+  # Allow access to font infrastructure
+  #include <abstractions/font>
```

Comment:
I'm using Ubuntu 20.04, and it complained at this line. Got it to work with
```suggestion
  #include <abstractions/fonts>
```
Will this work across distributions?

Reply:
That looks like a mis-key on my part, will get that updated.